docs/application/dotnet/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1096 @@
+    {
+        /// Your code
+    }
+
+    void SetUnsetPreRecognitionResultUpdatedCb()
+    {
+        try
+        {
+            /// Set the callback
+            VoiceControlManagerClient.PreRecognitionResultUpdated += preRecognitionResultUpdated;
+
+            /// Unset the callback
+            VoiceControlManagerClient.PreRecognitionResultUpdated -= preRecognitionResultUpdated;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
+-   Set the all recognition result received callback invoked when voice control engine sends the all recognition result to voice control manager.
+
+    In the callback, the recognized result, recognized text, and engine message results are included and recognized result can include more than two voice commands if two voice commands those have same command string are registered by each voice control clients.
+
+    If you want to select specific command in the recognized result, you can use SetRecognizedCommandsSelectionDelegate() method.
+
+    ```csharp
+    /// callback
+    void allRecognitionResultReceived(object sender, AllRecognitionResultEventArgs e)
+    {
+        /// Your code
+    }
+
+    void SetUnsetAllRecognitionResultReceivedCb()
+    {
+        try
+        {
+            /// Set the callback
+            VoiceControlManagerClient.AllRecognitionResultReceived += allRecognitionResultReceived;
+
+            /// Unset the callback
+            VoiceControlManagerClient.AllRecognitionResultReceived -= allRecognitionResultReceived;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
+-   Set the recognition result updated callback invoked when voice control engine set the recognition result updated to voice control manager.
+
+    You can get the recognized result, recognized text, and recognized command list in this callback.
+
+    If no commands are matched, the callback returns `RecognizedResult.Rejected`:
+
+    ```csharp
+    /// callback
+    void recognitionResultUpdated(object sender, RecognitionResultUpdatedEventArgs e)
+    {
+        /// Your code
+    }
+
+    void SetUnsetRecognitionResultUpdatedCb()
+    {
+        try
+        {
+            /// Set the callback
+            VoiceControlManagerClient.RecognitionResultUpdated += recognitionResultUpdated;
+
+            /// Unset the callback
+            VoiceControlManagerClient.RecognitionResultUpdated -= recognitionResultUpdated;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
+-   Set the recognized commands selection callback invoked when the voice control manager select specific voice commands in all recognized result:
+
+    When utterence is recognized, all recognized result can include more than two voice commands if more than two voice commands those have same command string are registered by each voice control clients.
```

Comment:
@sungraejo Suggesting a small tweak in the sentence, by breaking it into two sentences for better reading flow, as: 

When utterance is recognized, all recognized result can include more than two voice commands. If more than two voice commands that have same command string, are registered by each voice control clients.

Please verify.

Reply:
@mijin85cho, line 447 must have a period instead of a semi colon in the end.

Set the recognized commands selection callback invoked when the voice control manager select specific voice commands in all recognized result.

Reply:
Thanks for checking @surbhig12

However, the line I hoped you check was line 449. Sorry for the confusion.
I think fixing the typo is needed. But cutting the sentence into two before "If more.." is not recommended in my thought.

Reply:
@mijin85cho, I too agree with your point and suggest rephrasing to:

When utterance is recognized, all the recognized result can include more than two voice commands if the two voice commands have the same command string that are registered by each voice control clients.

Please let me know your views.

Reply:
@surbhig12 Suggestion. 
As it is saying that the several registered voice commands that have same string can cause several voice commands recognition result,
I think leaving "two voice commands" only to the second phrase can change the meaning slightly and doesn't match with the previous phrase, "more than two voice commands".

When utterance is recognized, all the recognized result can include **more than two voice commands** if the **two voice commands** have the same command string that are registered by each voice control clients.

So, how about below?

When utterance is recognized, all the recognized result can include more than two voice commands if more than two voice commands that have the same command string are registered by each voice control clients.

Reply:
@mijin85cho, Please let me know if this is what we are trying to say so that I can rephrase better:

When utterance is recognized, all the recognized result can include more than two voice commands if these commands have the same command string that is registered by each voice control clients.

Reply:
@surbhig12 I think this suggestion is good. Thanks for the modification.

@sungraejo Would you please refer to the conversation and apply the suggestion?

When utterance is recognized, all the recognized result can include more than two voice commands if these commands have the same command string that is registered by each voice control clients.